Firebase/Auth/Source/FIRAuth.m

```diff
@@ -554,6 +563,40 @@ - (void)signInWithEmail:(NSString *)email
   }];
 }
 
+- (void)signInAndRetrieveDataWithEmail:(NSString *)email
+                              password:(NSString *)password
+                            completion:(FIRAuthDataResultCallback)completion {
+  dispatch_async(FIRAuthGlobalWorkQueue(), ^{
+  FIRAuthDataResultCallback decoratedCallback =
+      [self signInFlowAuthDataResultCallbackByDecoratingCallback:completion];
+    [self internalSignInAndRetrieveDataWithEmail:email
+                                        password:password
+                                      completion:^(FIRAuthDataResult *_Nullable authResult,
```

Comment:
Just to pass in `decoratedCallback` directly?

Reply:
Yes, done.